web_poet/pages.py

```diff
@@ -59,7 +59,12 @@ class ItemPage(Injectable, Returns[ItemT]):
 
     def __init_subclass__(cls, skip_nonitem_fields: bool = False, **kwargs):
         super().__init_subclass__(**kwargs)
-        cls._skip_nonitem_fields = skip_nonitem_fields
+
+        if getattr(cls, "_skip_nonitem_fields", None) is True:
+            # See: https://github.com/scrapinghub/web-poet/issues/141
+            cls._skip_nonitem_fields = True
```

Comment:
Attrs with `slot=True` causes this to run twice where the second call would always have `skip_nonitem_fields=False`. This workaround prevents overriding the value of `True` when it was intentionally passed earlier.

Reply:
That's a super weird looking code: we're checking if value is True, and if it's True, we set it to True :) Magic!

Does it work when subclass sets skip_nonitem_fields to False, not to True, while the base class has skip_nonitem_fields=True? Could you please add a test for this?

Reply:
Good catch! It wasn't working before. Addressed in f829a95.